gateway/src/commonMain/kotlin/builder/PresenceBuilder.kt

```diff
@@ -36,7 +40,14 @@ public class PresenceBuilder {
         game = DiscordBotActivity(name, ActivityType.Competing)
     }
 
-    public fun toUpdateStatus(): UpdateStatus = UpdateStatus(since, listOfNotNull(game), status, afk)
+    public fun custom(state: String) {
+        game = DiscordBotActivity(name = "Custom Status", state = Optional(state), type = ActivityType.Custom)
+    }
```

Comment:
I really don't like the API design on adding an extra parameter to all the other options, I do get your point, but imo the API design is better with a top-level parameter, if it is really that important to do that we can add a runtime check whether game/state is set

Reply:
hm, what abou just having `public var state` (and no `fun custom`) and then in the build methods treating it as custom status when `state` is set and `game` is missing? otherwise if `state` and `game` are there, just add them together?

Reply:
I still like having a custom function more, but I can't understand why you wouldn't want that and this keeps the core of the imo better api, so yeah that would be ok with me

Reply:
i mean yeah, we could have that function, it would just look like this then:

```kotlin
fun custom(state: String) {
    this.state = state
}
```

Reply:
can be added later if needed